clevertap-core/src/main/java/com/clevertap/android/sdk/InAppNotificationActivity.java

```diff
@@ -64,6 +61,14 @@ public interface PushPermissionResultCallback {
 
     public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        getOnBackPressedDispatcher().addCallback(this, new OnBackPressedCallback(true) {
+            @Override
+            public void handleOnBackPressed() {
+                finish();
+                didDismiss(null);
+            }
+        });
```

Comment:
do we not have to remove callback?

Reply:
@CTLalit This is the documentation

```
Receive callbacks to a new OnBackPressedCallback when the given LifecycleOwner is at least started.
This will automatically call addCallback(OnBackPressedCallback) and remove the callback as the lifecycle state changes. As a corollary, if your lifecycle is already at least started, calling this method will result in an immediate call to addCallback(OnBackPressedCallback).
When the LifecycleOwner is destroyed, it will automatically be removed from the list of callbacks. The only time you would need to manually call OnBackPressedCallback.remove() is if you'd like to remove the callback prior to destruction of the associated lifecycle.
If the Lifecycle is already destroyed when this method is called, the callback will not be added.
```